src/impy/kinematics.py

```diff
@@ -26,7 +89,10 @@ class CompositeTarget(object):
     Examples of such composite targets are Air, CO_2, HCl, C_2H_60.
     """
 
-    def __init__(self, label=""):
+    from numpy.random import default_rng
+
+    def __init__(self, component_list, label=""):
+        self.rng = self.default_rng()
```

Comment:
Please add an optional keyword `random_state` which allows the user to pass their own random number generator. If `random_state` is None, use `np.random.default_rng()`. If the user provides an int, use that to seed the `np.random.default_rng`. See https://github.com/scikit-hep/resample/blob/develop/src/resample/bootstrap.py#L34 for an example. This is how scipy does it.

Reply:
It is important to be able to set the seed, because we want deterministic randomness. This is even more flexible, because it allows users to pass their preferred random number generator instead of the numpy default. Moreover, it allows one to pass a mocked RNG for unit tests.

Reply:
Good idea. Added.